index/server/main.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/deislabs/oras/pkg/content"
+	"github.com/deislabs/oras/pkg/oras"
+	indexSchema "github.com/devfile/registry-support/index/generator/schema"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/gin-gonic/gin"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	devfileName            = "devfile.yaml"
+	devfileConfigMediaType = "application/vnd.devfileio.devfile.config.v2+json"
+	devfileMediaType       = "application/vnd.devfileio.devfile.layer.v1"
+	scheme                 = "http"
+	registryService        = "localhost:5000"
+)
+
+var (
+	stacksPath = os.Getenv("DEVFILE_STACKS")
+	indexPath  = os.Getenv("DEVFILE_INDEX")
+)
+
+func main() {
+	// Wait until registry is up and running
+	isDone := false
+	for !isDone {
+		resp, err := http.Get(scheme + "://" + registryService)
+		if err != nil {
+			log.Fatal(err.Error())
```

Comment:
Looks like `Connection refused` errors from the registry server will also lead to `http.Get` returning an error, so I don't think we can safely do a log.Fatal here.

This is what I get if I change line 43 to log the error and continue:
```
2020/11/26 19:45:55 Get "http://localhost:5000": dial tcp [::1]:5000: connect: connection refused
```

Reply:
You are right, this shouldn't be a place to throw fatal error directly, I can make it fail until it reaches the timeout.